axoupdater/src/lib.rs

```diff
@@ -802,24 +803,48 @@ async fn get_github_releases(
     app_name: &str,
 ) -> AxoupdateResult<Vec<Release>> {
     let client = reqwest::Client::new();
-    let resp: Vec<Release> = client
-        .get(format!("{GITHUB_API}/repos/{owner}/{name}/releases"))
-        .header(ACCEPT, "application/json")
-        .header(
-            USER_AGENT,
-            format!("axoupdate/{}", env!("CARGO_PKG_VERSION")),
-        )
-        .send()
-        .await?
-        .error_for_status()
-        .map_err(|_| AxoupdateError::ReleaseNotFound {
-            name: name.to_owned(),
-            app_name: app_name.to_owned(),
-        })?
-        .json()
-        .await?;
+    let mut url = format!("{GITHUB_API}/repos/{owner}/{name}/releases");
+    let next_pattern = Regex::new(r#"/(?<=<)([\S]*)(?=>; rel="Next")"/i"#).expect("bad regex");
```

Comment:
could you add a comment/citation/example for what's up with this?

Reply:
for your own edification it was from the github API docs on their pagination: https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28. i can't use it because it has look aheads/behinds and the rust regex crate doesn't support that apparently.

this is what the header looks like and i suspect you'll see why its such a pain in the ass to parse:

```
link: <https://api.github.com/repositories/1300192/issues?per_page=2&page=2>; rel="next", <https://api.github.com/repositories/1300192/issues?per_page=2&page=7715>; rel="last"
```

anyways you can see my crimes in the commit now that just fully eliminate the regex usage